src/main/java/de/unistuttgart/iste/gits/content_service/service/ContentService.java

```diff
@@ -356,4 +370,51 @@ private List<SkillType> getSkillTypesOfChapter(final UUID chapterId) {
                 .distinct()
                 .toList();
     }
+
+
+    /**
+     * returns an ordered list of content with no links to any section for a list of chapter ID.
+     *
+     * @param chapterIds list of chapter IDs in which content is to be searched for
+     * @return ordered list of content that matches the order of the chapterID argument
+     */
+    public List<List<Content>> getContentWithNoSection(List<UUID> chapterIds) {
+        List<List<Content>> result = new ArrayList<>(chapterIds.size());
+
+        // get a list containing all sections for the given chapters, but not divided by chapter yet
+        List<SectionEntity> sectionEntities = sectionRepository.findByChapterIdIn(chapterIds);
+        List<ContentEntity> contentEntities = contentRepository.findByChapterIdIn(chapterIds);
+
+        // collect optional and required content in form of a stream for further processing
+        Stream<Set<ContentEntity>> requiredContentStream = sectionEntities.stream()
```

Comment:
this logic is already needed in another method as well (I think the suggestens and/or the user progress) -> maybe extract three the methods into the entity class, one for getting the required contents, one for getting the optional contents and one for both? this would reduce redundancy if you then use it in the other places as well.